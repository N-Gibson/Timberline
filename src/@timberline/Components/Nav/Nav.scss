header {
  align-items: center;
  background-color: #c8ad96;
  box-shadow: inset 0px 0px 20px 0px #00000040;
  display: flex;
  justify-content: space-between;
  height: 125px;
  position: fixed;
  width: 100vw;
  z-index: 20;
}

#timberline_logo {
  width: 250px;
  transition: transform 0.25s;
  padding: 1em;
  &:hover {
    transform: scale(1.05);
  }
}

#logo_25 {
  width: 100px;
  margin: 0 1em;
  transition: transform 0.25s;
  &:hover {
    transform: scale(1.05);
  }
}

#nav_buttons_container {
  align-self: flex-end;
  width: 100%;
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
}

.nav_button {
  border-radius: 10px 10px 0 0;
  min-width: 125px;
  height: 75px;
  transition: 0.25s;
  border: none;
  outline: none;
  background-color: transparent;
  font-size: 1.25em;
  &:hover {
    // transform: scale(1.1);
    background-color: white;
    border-bottom: solid 2px #c8ad96;
    cursor: pointer;
    box-shadow: 3px 0px 10px #00000040;
  }
}

.active_button {
  background-color: white;
}
