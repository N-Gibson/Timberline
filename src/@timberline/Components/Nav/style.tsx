import { makeStyles } from '@material-ui/core'

export const useStyles = makeStyles(() => ({
  appBar: {
    display: 'flex',
    'flex-direction': 'row',
    'justify-content': 'flex-start',
  },
  logo: {
    height: '80px',
    'margin-right': '50px',
  },
}))
