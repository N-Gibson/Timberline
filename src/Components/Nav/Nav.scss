header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #C8AD96;
  box-shadow: inset 0px 0px 20px 0px #00000040;
}

#timberline_logo {
  width: 175px;
  transition: transform .25s;
  padding: 1em;
  &:hover {
    transform: scale(1.05);
  }
}

#logo_25 {
  width: 70px;
  margin: 0 1em;
  transition: transform .25s;
  &:hover {
    transform: scale(1.05);
  }
}

#nav_buttons_container {
  align-self: flex-end;
  width: 100%;
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
}

.nav_button {
  border-radius: 10px 10px 0 0;
  min-width: 100px;
  height: 50px;
  transition: .25s;
  border: none;
  outline: none;
  background-color: transparent;
  margin: 0 5px;
  font-size: 1.05em;
  &:hover  {
    // transform: scale(1.1);
    background-color: white;
    border-bottom: solid 2px #C8AD96;
    cursor: pointer;
    box-shadow: 3px 0px 10px #00000040;
  }
}

.active_button {
  background-color: white;
}