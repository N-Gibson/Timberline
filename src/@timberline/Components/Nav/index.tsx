import React from 'react'
import { Link } from 'react-router-dom'
import './Nav.scss'
import timberline_logo from '@timberline/assets/timberline-const.png'
import timberline_logo_only from '@timberline/assets/timberline_logo_only.png'
import logo_25 from '@timberline/assets/25+_timberline.png'

interface NavProps {
  active: string
}

const Nav: React.FC<NavProps> = ({ active }) => {
  const idleButton = 'nav_button'
  const activeButton = 'nav_button active_button'

  return (
    <header>
      <Link to="/">
        <img
          id="timberline_logo"
          src={timberline_logo_only}
          alt="timberline logo"
        />
      </Link>
      <section id="nav_buttons_container">
        <Link to="/about">
          <button
            className={active === 'about' ? activeButton : idleButton}
            id="about_nav_button"
          >
            About
          </button>
        </Link>
        <Link to="/projects">
          <button
            className={active === 'projects' ? activeButton : idleButton}
            id="projects_nav_button"
          >
            Projects
          </button>
        </Link>
        <Link to="/contact-us">
          <button
            className={active === 'contact' ? activeButton : idleButton}
            id="contact_us_nav_button"
          >
            Contact Us
          </button>
        </Link>
      </section>
      <Link to="/about">
        <img id="logo_25" src={logo_25} alt="logo" />
      </Link>
    </header>
  )
}

export { Nav }
