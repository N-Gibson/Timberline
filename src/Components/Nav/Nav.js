import React, { useState } from "react";
import { Link } from "react-router-dom";
import "./Nav.scss";
import timberline_logo from "../../assets/timberline-const.png";
import logo_25 from "../../assets/25+_timberline_test.png";

export const Nav = () => {
  const idleButton = "nav_button";
  const activeButton = "nav_button active_button";

  const [active, setActive] = useState("home");

  return (
    <header>
      <Link to="/" onClick={() => setActive("home")}>
        <img id="timberline_logo" src={timberline_logo} alt="timberline logo" />
      </Link>
      <section id="nav_buttons_container">
        <Link to="/about">
          <button
            className={active === "about" ? activeButton : idleButton}
            id="about_nav_button"
            onClick={() => setActive("about")}
          >
            About
          </button>
        </Link>
        <Link to="/projects">
          <button
            className={active === "projects" ? activeButton : idleButton}
            id="projects_nav_button"
            onClick={() => setActive("projects")}
          >
            Projects
          </button>
        </Link>
        <Link to="/contact-us">
          <button
            className={active === "contact" ? activeButton : idleButton}
            id="contact_us_nav_button"
            onClick={() => setActive("contact")}
          >
            Contact Us
          </button>
        </Link>
      </section>
      <Link to="history" onClick={() => setActive("history")}>
        <img id="logo_25" src={logo_25} alt="logo" />
      </Link>
    </header>
  );
};
