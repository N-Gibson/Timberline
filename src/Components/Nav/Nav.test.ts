import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './Nav';

describe('Nav', () => {
  const wrapper = shallow(<Nav />);
  
  it('Nav component should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});