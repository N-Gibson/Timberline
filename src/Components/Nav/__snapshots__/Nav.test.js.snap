// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav component should match the snapshot 1`] = `
<header>
  <Link
    to="/"
  >
    <img
      alt="timberline logo"
      id="timberline_logo"
      src="timberline-const.png"
    />
  </Link>
  <section
    id="nav_buttons_container"
  >
    <Link
      to="/about"
    >
      <button
        className="nav_button"
        id="about_nav_button"
      >
        About
      </button>
    </Link>
    <Link
      to="/projects"
    >
      <button
        className="nav_button"
        id="projects_nav_button"
      >
        Projects
      </button>
    </Link>
    <Link
      to="/contact-us"
    >
      <button
        className="nav_button"
        id="contact_us_nav_button"
      >
        Contact Us
      </button>
    </Link>
  </section>
  <Link
    to="history"
  >
    <img
      alt="logo"
      id="logo_25"
      src="25+_timberline_test.png"
    />
  </Link>
</header>
`;
